### PR TITLE
Feature/jjdd8 hfv 33bm

### PR DESCRIPTION
**Pull request by dagiel**
Add/Edit/Delete employee done.
Update employee to be fixed (front-end side)

### CODE REVIEW

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

File: .gitignore [hunk 1]
```diff
@@ -31,3 +31,5 @@ hs_err_pid*
 /web/src/main/webapp/bootstrap/
 /web/src/main/webapp/test/
 sendgrid.env
+
+!.idea/*
```
Comment (rafalmisiak) [inline .gitignore]: dlaczego zanegowaleś .idea? tego katalogu nie może być w repo

File: web/src/main/java/com/infoshare/academy/highfive/dao/EmployeeDao.java [hunk 2]
```diff
@@ -1,33 +1,45 @@
...
-    entityManager.persist(employee);
-  }
+    public List<Employee> listAllEmployees() {
+        return this.em.createNamedQuery("Employee.findAll").getResultList();
+    }
+
+    public void update(Employee employee){ em.merge(employee); }
 
+  public Employee delete(Long id) {
+    Employee removedEmployee = getById(id).orElseThrow();
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/dao/EmployeeDao.java]: jeśli rzucasz w przypadku pustego to nie ma sensu poniżesz robić ifa :)

File: web/src/main/java/com/infoshare/academy/highfive/dao/TeamDao.java [hunk 3]
```diff
@@ -3,13 +3,44 @@
...
+
+    public void update(Team team) {
+        entityManager.merge(team);
+    }
+
+    public void edit() {
+    }
+
+    public Team delete(Long id) {
+        Team removedTeam = getById(id).orElseThrow();
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/dao/TeamDao.java]: jw

File: web/src/main/java/com/infoshare/academy/highfive/dto/view/EmployeeView.java [hunk 4]
```diff
@@ -25,21 +25,21 @@
...
-    this.firstName = firstName;
-    this.surname = surname;
-    this.hireDate = hireDate;
-    this.holidayEntitlement = holidayEntitlement;
-    this.additionalEntitlement = additionalEntitlement;
-    this.teamId = teamId;
-    this.roleId = roleId;
-  }
+//  public EmployeeView(Long Id, String firstName, String surname, LocalDate hireDate, int holidayEntitlement, int additionalEntitlement, Team teamId, Role roleId) {
+//    this.Id = Id;
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/dto/view/EmployeeView.java]: nie wrzucamy zakomentowanego kodu do repo

File: web/src/main/java/com/infoshare/academy/highfive/web/servlet/MainServlet.java [hunk 5]
```diff
@@ -20,33 +22,44 @@
...
+    dataModel.put("currentMonthTotal", vacationService.getStatistics().getCurrentMonthTotal());
+    dataModel.put("nextMonthTotal", vacationService.getStatistics().getNextMonthTotal());
+    dataModel.put("absentToday", vacationService.getStatistics().getAbsentToday());
+    dataModel.put("totalEmployees", employeeService.listAllSize());
+    dataModel.put("pendingRequests", vacationService.getStatistics().getPendingRequests());
+
+    try {
+      template.process(dataModel, writer);
+    } catch (
+      TemplateException e) {
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/web/servlet/MainServlet.java]: coś z formatowaniem nie pykło

File: web/src/main/java/com/infoshare/academy/highfive/web/servlet/employee/AddEmployeeServlet.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        }
+        dataModel.put("teams", teamService.listAll());
+        dataModel.put("pluginCssTemplate", "plugin-css-stylesheet.ftlh");
+        dataModel.put("pluginJsTemplate", "plugin-js-add-employee.ftlh");
+
+        LOGGER.info("User (manager) provided with new employee adding form.");
+
+        try {
+            template.process(dataModel, writer);
+        } catch (
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/web/servlet/employee/AddEmployeeServlet.java]: jw

File: web/src/main/java/com/infoshare/academy/highfive/web/servlet/employee/EditEmployeeServlet.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        dataModel.put("contentTemplate", "edit-employee.ftlh");
+        dataModel.put("title", "List Employees");
+        dataModel.put("pluginCssTemplate", "plugin-css-edit-employee.ftlh");
+        dataModel.put("pluginJsTemplate", "plugin-js-edit-employee.ftlh");
+        dataModel.put("employees", employeeService.listAll());
+        dataModel.put("teams", teamService.listAll());
+
+        try{
+            template.process(dataModel, writer);
+        } catch (
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/web/servlet/employee/EditEmployeeServlet.java]: jw

File: web/src/main/java/com/infoshare/academy/highfive/web/servlet/employee/EmployeeListServlet.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        dataModel.put("method", req.getMethod());
+        dataModel.put("contentTemplate", "all-employee.ftlh");
+        dataModel.put("title", "List Employees");
+        dataModel.put("pluginCssTemplate", "plugin-css-stylesheet.ftlh");
+        dataModel.put("pluginJsTemplate", "plugin-js-servlets.ftlh");
+        dataModel.put("employees", employeeService.listAll());
+
+        try{
+            template.process(dataModel, writer);
+        } catch (
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/web/servlet/employee/EmployeeListServlet.java]: jw

File: web/src/main/java/com/infoshare/academy/highfive/web/servlet/team/AddTeamServlet.java [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+        } else {
+            dataModel.put("title", "Add team");
+            dataModel.put("action", "add");
+        }
+        dataModel.put("pluginCssTemplate", "plugin-css-stylesheet.ftlh");
+        dataModel.put("pluginJsTemplate", "plugin-js-add-team.ftlh");
+
+        try {
+            template.process(dataModel, writer);
+        } catch (
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/web/servlet/team/AddTeamServlet.java]: jw

File: web/src/main/java/com/infoshare/academy/highfive/web/servlet/team/EditTeamServlet.java [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+        dataModel.put("method", req.getMethod());
+        dataModel.put("contentTemplate", "edit-team.ftlh");
+        dataModel.put("title", "List Teams");
+        dataModel.put("pluginCssTemplate", "plugin-css-edit-team.ftlh");
+        dataModel.put("pluginJsTemplate", "plugin-js-edit-team.ftlh");
+        dataModel.put("teams", teamService.listAll());
+
+        try {
+            template.process(dataModel, writer);
+        } catch (
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/web/servlet/team/EditTeamServlet.java]: jw

File: web/src/main/java/com/infoshare/academy/highfive/web/servlet/team/TeamListServlet.java [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+        dataModel.put("method", req.getMethod());
+        dataModel.put("contentTemplate", "all-team.ftlh");
+        dataModel.put("title", "List Teams");
+        dataModel.put("pluginCssTemplate", "plugin-css-stylesheet.ftlh");
+        dataModel.put("pluginJsTemplate", "plugin-js-servlets.ftlh");
+        dataModel.put("teams", teamService.listAll());
+
+        try {
+            template.process(dataModel, writer);
+        } catch (
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/web/servlet/team/TeamListServlet.java]: jw

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for original commit.

**Comment by isacademyci** [comment]
Build started for original commit.

**Comment by isacademyci** [comment]
Build finished.

File: .gitignore (hunk 1 above)
Comment (dagiel) [reply .gitignore]: .idea usunięta z repo

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Pull request merged**